core/src/components/button/test/basic/button.e2e.ts

```diff
@@ -10,3 +10,28 @@ test.describe('button: basic', () => {
     expect(await page.screenshot({ fullPage: true })).toMatchSnapshot(`button-diff-${page.getSnapshotSettings()}.png`);
   });
 });
+
+test.describe('button: ripple effect', () => {
+  test('should not have visual regressions', async ({ page }, testInfo) => {
+    test.skip(testInfo.project.metadata.mode !== 'md', 'Ripple effect is only available in md mode');
+
+    await page.setContent(`
+      <ion-button>Button</ion-button>
+      <ion-button id="rippleBtn">Button with Ripple</ion-button>
+    `);
+
+    const button = page.locator('#rippleBtn');
+    const boundingBox = await button.boundingBox();
+
+    if (boundingBox) {
+      await page.mouse.move(boundingBox.x + boundingBox.width / 2, boundingBox.y + boundingBox.height / 2);
+    }
+
+    await page.mouse.down();
+    await page.waitForTimeout(100);
```

Comment:
What if you waited for the `.ripple-effect` element to be inside `ion-ripple-effect` in the DOM? That might give you more consistent results: https://github.com/ionic-team/ionic-framework/blob/780f16d9e04ee5aaaf91bb7c6ef15c72cc8aeb45/core/src/components/ripple-effect/ripple-effect.tsx#L56